postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -519,27 +518,29 @@ def _get_db_explain_setup_state(self, dbname):
         # type: (str) -> Tuple[Optional[DBExplainError], Optional[Exception]]
         try:
             self.db_pool.get_connection(dbname, self._conn_ttl_ms)
-        except psycopg2.OperationalError as e:
+        except psycopg.OperationalError as e:
             self._log.warning(
                 "cannot collect execution plans due to failed DB connection to dbname=%s: %s", dbname, repr(e)
             )
             return DBExplainError.connection_error, e
-        except psycopg2.DatabaseError as e:
+        except psycopg.DatabaseError as e:
             self._log.warning(
                 "cannot collect execution plans due to a database error in dbname=%s: %s", dbname, repr(e)
             )
             return DBExplainError.database_error, e
+        finally:
+            self.db_pool.set_conn_inactive(dbname)
```

Comment:
This is decoupling the connection getting from its context manager, which seems unnecessary. This can be accomplished by replacing:

```python
try:
    self.db_pool.get_connection(dbname, self._conn_ttl_ms)
except:
    ...
```

with 

```python
try:
    with self.db_pool.get_connection(dbname, self._conn_ttl_ms):
        pass
except:
    ...
```

The reason I prefer this is because it keeps the API tighter and more defensible, making it harder to introduce flows that would result in the inactivity not getting called.

Reply:
This is actually a mistake, and was part of the original implementation I had. If you look at the other uses, I am using a context manager in all other places. 

Thanks for catching it, I'll fix it!